### PR TITLE
Upgrade to JDK11

### LINKED ISSUE #2637: Upgrade to Java 11
The new LTS version is out.
It is the good time to follow the rule : target the previous LTS.

WDYT @krmahadevan ?

### PR DESCRIPTION
**Pull request by krmahadevan**
* Also upgraded all the dependencies to the latest
* Fixed the matrix as well to toggle between 11/16/17

Fixes #2637

### Did you remember to?

- [ ] Add test case(s)
- [X] Update `CHANGES.txt`
- [X] Auto applied styling via `./gradlew autostyleApply`

We encourage pull requests that:

* Add new features to TestNG (or)
* Fix bugs in TestNG

If your pull request involves fixing SonarQube issues then we would suggest that you please discuss this with the 
[TestNG-dev](https://groups.google.com/forum/#!forum/testng-dev) before you spend time working on it.

**Note:** For more information on contribution guidelines  please make sure you refer our [Contributing](.github/CONTRIBUTING.md) section for detailed set of steps.

### CODE REVIEW

File: settings.gradle.kts [hunk 1]
```diff
@@ -6,7 +6,15 @@ rootProject.name = "testng-root"
 
 plugins {
     `gradle-enterprise`
-    id("de.fayard.refreshVersions") version "0.10.0"
+    id("de.fayard.refreshVersions") version "0.30.1"
+////                            # available:"0.10.1"
```
Comment (juherr) [inline settings.gradle.kts:10]: Remove old commented versions

File: versions.properties [hunk 2]
```diff
@@ -11,16 +11,61 @@ plugin.com.github.johnrengelman.shadow=7.0.0
 
 plugin.com.github.vlsi.gradle-extensions=1.74
 
-plugin.com.github.vlsi.stage-vote-release=1.74
+plugin.com.github.vlsi.stage-vote-release=1.77
+##                            # available=1.75
+##                            # available=1.76
+##                            # available=1.77
```
Comment (juherr) [inline versions.properties:17]: Remove old commented version

**Review by juherr — CHANGES_REQUESTED** [review]
Commented version are just here to show you all available new versions. When you upgrade you are supposed to remove old versions.

File: settings.gradle.kts (hunk 1 above)
Comment (krmahadevan) [reply settings.gradle.kts]: Thanks @juherr - I didnt know that. Fixed

File: versions.properties (hunk 2 above)
Comment (krmahadevan) [reply versions.properties]: Thanks @juherr - I didnt know that. Fixed

**Comment by krmahadevan** [comment]
@juherr - Can you please take a quick look once again ? I have fixed the review comments.

File: .github/workflows/matrix.js [hunk 3]
```diff
@@ -8,28 +8,28 @@ matrix.addAxis({
   title: x => x.version + ', ' + x.group,
   values: [
     // Zulu
-    {group: 'Zulu', version: '8', distribution: 'zulu', jit: 'hotspot'},
     {group: 'Zulu', version: '11', distribution: 'zulu', jit: 'hotspot'},
     {group: 'Zulu', version: '16', distribution: 'zulu', jit: 'hotspot'},
```
Comment (juherr) [inline .github/workflows/matrix.js:12]: You can remove java16 as it doesn't have any support https://en.wikipedia.org/wiki/Java_version_history

File: versions.properties [hunk 4]
```diff
@@ -82,18 +60,11 @@ version.org.ops4j.pax.url..pax-url-aether=2.6.1
 version.org.spockframework..spock-core=1.0-groovy-2.4
 
 version.org.webjars..jquery=3.5.1
-##              # available=3.6.0
```
Comment (juherr) [inline versions.properties:85]: Here you have to choose between upgrade to 3.6.0 or keep the new version reminder.

**Review by juherr — CHANGES_REQUESTED** [review]

**Comment by krmahadevan** [comment]
@juherr - I have fixed both the review comments. Please check

File: .github/workflows/matrix.js [hunk 5]
```diff
@@ -144,7 +134,7 @@ include.forEach(v => {
     // so it might reveal missing synchronization in TestNG code
     v.name += ', stress JIT';
     jvmArgs.push('-XX:+UnlockDiagnosticVMOptions');
-    if (v.jdk.version >= 8) {
+    if (v.jdk.version >= 11) {
```
Comment (juherr) [inline .github/workflows/matrix.js:137]: You can remove the condition because it is always true now.

**Review by juherr — APPROVED** [review]
Just fix the last issue and merge 👍

File: .github/workflows/matrix.js (hunk 5 above)
Comment (krmahadevan) [reply .github/workflows/matrix.js]: Fixed and pushed the amended commit.

**Pull request merged**